### PR TITLE
Add X509Extension parsing

### PR DESCRIPTION
**Pull request by gordonwang0**

### CODE REVIEW

File: openssl2/src/extension.rs [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+            } else {
+                return None;
+            }
+        }
+
+        Some(ext_key_usage)
+    }
+
+    /// Check if this `extKeyUsage` has serverAuth set.
+    pub fn server_auth(&self) -> bool {
```
Comment (Arnavion) [inline openssl2/src/extension.rs:52]: Could just make it a `pub` field.

File: openssl2/src/extension.rs [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+#[derive(Debug)]
+pub struct ExtKeyUsage {
+    /// Server authentication.
+    server_auth: bool,
+    // extKeyUsage contains other fields, but only serverAuth is used in Certs Service.
+    // Unused other fields are omitted.
+}
+
+impl ExtKeyUsage {
+    /// Parse an extension as `extKeyUsage`.
```
Comment (Arnavion) [inline openssl2/src/extension.rs:13]: Might be useful to add the URLs of the RFCs and sections that document the structure.

**Review by Arnavion — APPROVED** [review]

**Review by arsing — APPROVED** [review]

File: openssl2/src/extension.rs (hunk 1 above)
Comment (gordonwang0) [reply openssl2/src/extension.rs:52]: Fixed

File: openssl2/src/extension.rs (hunk 2 above)
Comment (gordonwang0) [reply openssl2/src/extension.rs:16]: Fixed

**Pull request merged**